src/bp/core/services/dialog/instruction/strategy.ts

```diff
@@ -167,8 +173,7 @@ export class ActionStrategy implements InstructionStrategy {
       const service = await this.actionService.forBot(botId)
       await service.runAction({
         actionName,
-        actionCode:
-          code && `const { ${argsToConst(variables.map(x => x.name))} } = event.state.workflow.variables\n${code}`,
+        actionCode: code && `const { ${argsToConst(variables.map(x => x.name))} } = workflow.variables\n${code}`,
```

Comment:
did you mean to use `currentWorkflow` here ? if not what's the difference ? and I can't see where `workflow` is defined

Reply:
`event.state.workflow` is dynamic and will always point to the current workflow (in state manager)
```js
Object.defineProperty(state, 'workflow', {
      get() {
        return state.session.workflows?.[state.session.currentWorkflow!]
      },
      configurable: true
    })
```